Jenkinsfile.integration

```diff
@@ -24,6 +24,10 @@ pipeline {
     parameters {
         /* first value in the list is the default */
         choice(choices: ['airship', 'osh'], description: 'Which deployment mechanism?', name: 'deployment')
+
+        booleanParam(name: 'run_tempest', defaultValue: true, description: 'Run tempest tests?')
+
+        choice(choices: ['full', 'smoke'], description: 'Run full or smoke tempest tests?', name: 'tempest_test_type')
```

Comment:
Its added as an option as currently tempest tests execution adds to already a lengthy airship deployment which in turn limits the frequency of running this job. Also a very long job execution adds to infra resource constraints as a single job will take longer time. 
As per my previous smoke tests execution, it adds around 1 hour and full tests execution adds somewhere between 2-3 hours in local deployment. 
I understand that this is not a desirable behavior and we need to improve it. 
I was thinking that this way we can run jobs with tempest less frequently but on periodic basis so that we have visibility of  issues. And in parallel, we can also work on improving airship deployment and tempest test execution time frames.

Reply:
For other part of having blacklist file, its there for tempest execution but possibly, tempest play logic can be enhanced to be provided as external input file. I will think that will be something needs to be done in separate review when needed.

https://github.com/SUSE-Cloud/socok8s/blob/master/playbooks/roles/airship-deploy-tempest/files/tempest_blacklist

Usage: https://github.com/SUSE-Cloud/socok8s/blob/master/site/soc/software/charts/osh/openstack-tempest/tempest.yaml#L37

I am not sure about exposing this part in tempest job as configurable parameter. May be this can be revisited later when we have better idea about different scenarios and support in tempest charts.

Reply:
> Its added as an option as currently tempest tests execution adds to already a lengthy airship deployment which in turn limits the frequency of running this job. Also a very long job execution adds to infra resource constraints as a single job will take longer time.
> As per my previous smoke tests execution, it adds around 1 hour and full tests execution adds somewhere between 2-3 hours in local deployment.
> I understand that this is not a desirable behavior and we need to improve it.
> I was thinking that this way we can run jobs with tempest less frequently but on periodic basis so that we have visibility of issues. And in parallel, we can also work on improving airship deployment and tempest test execution time frames.

There are two things we need to improve here: quality and effiency.

For me, adding one hour for a tempest smoke test is ... insane. That shouldn't be the case.
You're right, that's too long.

This highlights issues. We may have a problem of performance, which would still apply to customers, so we need to fix this.

In the meantime, what I am proposing helps you. It basically allows a file whitelisting what to do. In some specific testing, we could edit the file to increase the coverage (like a scenario for refstack, adding all the tests necessary).
In other cases, devs will just add their test into the list, and improve coverage over time. We can then decide in the review if this should be in a specific scenario or not.

That sounds like a win win to me: Very limited scope, with ability to extend.

So I see no reason to allow the skip of testing.

Reply:
hesitating to add to already long conversation here. but we have both smoke test and full tempest here. so maybe smoke test for the gate, full test for the daily?

Reply:
I see there are multiple aspects to this which needs to be figured out. 
1) Currently socok8s don't have mechanism to limit (include) certain set of tests via **external** input file. This might be relatively easy change but we may need to make them in upstream osh charts first.
2) Then there is task of identifying which set of tests are essential and how much time they take.
2 b) And then need to identify scenarios to run selectively based on service/ component change. This more seems like bigger task and may not have priority among many other things we have on our plate.

3) Then we need to improve the workflow around tempest charts and how the are invoked currently in socok8s. I think this is something we can do upstream and will provide more bang for buck here.

But currently we are not there yet and in meantime, imho having a mechanism to run tempest in CI selectively for tests visibility can be useful. And that's why this flag.

For other changes, I will suggest to create stories and address changes in separate set of PRs.

Reply:
I agree with @kantarun's summary here.

At the moment we aren't exercising any tempest tests in the gate (or otherwise?), so we should focus on making that happen. The rest is optimisation that can be a followup in my opinion.

Reply:
I just want to remove the ability to disable tests.

We are testing quality with linting at all times, and so should we run tempest every time we run functional testing.

Do we really want to not check what we deliver?

Reply:
I think we agree then. Let's remove the conditionals, and I am good.

Reply:
I do agree with the universal idea that we should be running functional test with PR change. 

But in current state of things, we are not there yet. So I have more pragmatic opinion about running tempest periodically rather than running for every non-doc PR change. This adds to a already long-running job which adds to already-constrained ECP resource consumption and unreliability of gate job deployment.

That being said, I will go along with consensus here so I can keep or remove the flag based on it. 
FYI, run_tempest flag default is true and we do have job definitions in automation repo where we are planing to run tempest job every 8 hrs (it takes 4-5 hrs to run) .